src/infra/dataverse/sparql/sparqlGateway.ts

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable max-lines-per-function */
+import * as A from 'fp-ts/lib/Array'
+import { flow, pipe } from 'fp-ts/lib/function'
+import * as O from 'fp-ts/lib/Option'
+import * as TE from 'fp-ts/TaskEither'
+import type { DataverseEntity } from '@/domain/dataverse/entity'
+import type {
+  DataversePort,
+  RetrieveDataverseQueryFilters,
+  RetrieveDataverseResult
+} from '@/domain/dataverse/port'
+import type { SparqlBinding, SparqlResult } from './dto'
+
+export const sparqlGateway: DataversePort = {
+  retrieveDataverse: (
+    language: string,
+    limit: number,
+    offset: number,
+    filters: RetrieveDataverseQueryFilters
+  ): TE.TaskEither<Error, RetrieveDataverseResult> => {
+    const query = `
+      PREFIX core: <https://ontology.okp4.space/core/>
+      PREFIX owl: <http://www.w3.org/2002/07/owl#>
+      PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+      PREFIX serviceMetadata: <https://ontology.okp4.space/metadata/service/>
+      PREFIX datasetMetadata: <https://ontology.okp4.space/metadata/dataset/>
+      PREFIX dataspaceMetadata: <https://ontology.okp4.space/metadata/dataspace/>
+      SELECT ?id ?metadata ?title ?description ?type
+      WHERE {
+        {
+          ?id rdf:type core:Service .
+          ?metadata rdf:type serviceMetadata:GeneralMetadata .
+        } UNION {
+          ?id rdf:type core:Dataset .
+          ?metadata rdf:type datasetMetadata:GeneralMetadata .
+        } UNION {          
+          ?id rdf:type core:DataSpace .
+          ?metadata rdf:type dataspaceMetadata:GeneralMetadata .
+        }
+        ${filters.byType !== 'all' ? `FILTER ( contains(str(?type), "${filters.byType}" ))` : ''}
+        ?id rdf:type ?type .
+        ?type rdf:type owl:Class .
+        ?metadata core:describes ?id .
+        ?metadata core:hasTitle ?title .
+        ?metadata core:hasDescription ?description .
+        FILTER ( langMatches(lang(?title),'${language}') && langMatches(lang(?description),'${language}') )
+      }
+      LIMIT ${limit}
+      OFFSET ${offset}
+`
+
+    const fetchHeaders: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Authorization: `Basic ${APP_ENV.sparql['credentials']}`,
```

Comment:
Credentials are optional. Remove `Authorization` header if no credentails provided.

Reply:
As ontology DB is now opened with a read-only access for anonymous, I will remove `Authorization ` header as well as `env` variables.